src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs

```diff
@@ -48,12 +48,31 @@ public static bool IsBuiltInAnalyzer(this DiagnosticAnalyzer analyzer)
             return analyzer is IBuiltInAnalyzer || analyzer.IsWorkspaceDiagnosticAnalyzer() || analyzer.IsCompilerAnalyzer();
         }
 
-        public static bool ShouldRunForFullProject(this DiagnosticAnalyzerService service, DiagnosticAnalyzer analyzer, Project project)
+        public static bool IsOpenFileOnly(this DiagnosticAnalyzer analyzer, Workspace workspace)
         {
             var builtInAnalyzer = analyzer as IBuiltInAnalyzer;
             if (builtInAnalyzer != null)
             {
-                return !builtInAnalyzer.OpenFileOnly(project.Solution.Workspace);
+                return builtInAnalyzer.OpenFileOnly(workspace);
+            }
+
+            return false;
+        }
+
+        public static bool ShouldRunForFullProject(this DiagnosticAnalyzerService service, DiagnosticAnalyzer analyzer, Project project)
+        {
+            if (analyzer.IsBuiltInAnalyzer())
```

Comment:
I think the method name is confusing. will change it as IsAnalyzerWithNonHiddenDescriptor